online/src/main/scala/ai/chronon/online/MetadataStore.scala

```diff
@@ -139,6 +139,44 @@ class MetadataStore(kvStore: KVStore, val dataset: String = ChrononMetadataKey,
     },
     { join => Metrics.Context(environment = "join.meta.fetch", join = join) })
 
+  // Validate whether the join exists in the active groupBy list saved in kv store
+  def validateJoinExist(team: String, name: String): Boolean = {
+    val activeJoinList: Try[Seq[String]] = getJoinListByTeam(team)
+    if (activeJoinList.isFailure) {
+      logger.error(s"Failed to fetch active join list for team $team")
+      false
+    } else {
+      val joinKey = "joins/" + name
+      if (activeJoinList.get.contains(joinKey)) {
+        true
+      } else {
+        logger.error(s"Join $name not found in active join list for team $team")
+        false
+      }
+    }
+  }
+
+  // Validate whether the groupBy exists in the active groupBy list saved in kv store
+  def validateGroupByExist(team: String, name: String): Boolean = {
+    val activeGroupByList: Try[Seq[String]] = getGroupByListByTeam(team)
+    if (activeGroupByList.isFailure) {
+      logger.error(s"Failed to fetch active group_by list for team $team")
+      false
+    } else {
+      val groupByKey = if (name.contains("/")) {
+        "group_bys/" + name
+      } else {
+        "group_bys/" + name.replaceFirst("\\.", "/")
+      }
```

Comment:
why do we have different regex rules for group by and join?

Reply:
@pengyu-hou For join the fetch is always through the fetchJoin. For groupBy the fetch could be through the groupBy fetch directly or through the join part fetch inside a join. For the first situation the name is the name user sent to the fetcher client, like`zipline_test/test_online_group_by_small.v2` for the second situation the name is the join part name, like `zipline_test.test_online_group_by_small.v2` I have to use this work around to normalize the name here.

Reply:
I see.  

Could you add some comments to the code?